napari/components/viewer_model.py

```diff
@@ -409,6 +410,43 @@ def add_pyramid(self, pyramid, *args, **kwargs):
         self.add_layer(layer)
         return layer
 
+    def add_volume(self, volume, *args, **kwargs):
+        """Adds an volume layer to the layers list.
+
+        Parameters
+        ----------
+        pyramid : list
+            3D array volume data. If the last dimensions of the images have length 3
+            or 4 they can be interpreted as RGB or RGBA if multichannel is
+            `True`.
+        metadata : dict, optional
+            Image metadata.
+        multichannel : bool, optional
+            Whether the image is multichannel RGB or RGBA if multichannel. If
+            not specified by user and the last dimension of the data has length
+            3 or 4 it will be set as `True`. If `False` the image is
+            interpreted as a luminance image.
+        colormap : str, vispy.Color.Colormap, 2-tuple, dict, optional
+            Colormap to use for luminance images. If a string must be the name
+            of a supported colormap from vispy or matplotlib. If a tuple the
+            first value must be a string to assign as a name to a colormap and
+            the second item must be a Colormap. If a dict the key must be a
+            string to assign as a name to a colormap and the value must be a
+            Colormap.
+
+        name : str, keyword-only
+            Name of the layer.
+
+        Returns
+        -------
+        layer : :class:`napari.layers.Volume`
+            The newly-created volume layer.
+        """
+        layer = layers.Volume(volume, *args, **kwargs)
```

Comment:
here you need to add
```python
if self.dims.ndim == 2:
    self.dims.ndim = 3
```
to boost the number of dimensions so that setting the display `-3` is correct

Reply:
fixed